modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/K6ClientCodegen.java

```diff
@@ -744,9 +886,277 @@ private static String getAccept(OpenAPI openAPI, Operation operation) {
+                && !requestBody.getContent().get(contentTypeValue).getExamples().isEmpty());
+    }
+
+    /**
+     * Extract example for given request body.
+     *
+     * @param requestBody
+     * @param contentTypeValue
+     * @param bodyOrFormParams
+     */
+    private void extractRequestBodyExample(RequestBody requestBody, String contentTypeValue,
+            List<Parameter> bodyOrFormParams) {
+
+        Optional<Map.Entry<String, Example>> requestBodyExampleEntry = requestBody.getContent().get(contentTypeValue)
+                .getExamples().entrySet().stream().findFirst();
+
+        if (requestBodyExampleEntry.isPresent()) {
+
+            Example requestBodyExample = requestBodyExampleEntry.get().getValue();
+
+            try {
+                JsonNode requestBodyExampleValueJsonNode = Json.mapper()
+                        .readTree(String.valueOf(requestBodyExample.getValue()));
+
+                Iterator<Map.Entry<String, JsonNode>> fields = requestBodyExampleValueJsonNode.fields();
+                while (fields.hasNext()) {
+                    Map.Entry<String, JsonNode> fieldsEntry = fields.next();
+
+                    JsonNode exampleValueAsJsonNode = fieldsEntry.getValue();
+
+                    Parameter k6Parameter = new Parameter(fieldsEntry.getKey(),
+                            exampleValueAsJsonNode.isNumber() ? exampleValueAsJsonNode.asText()
+                                    : exampleValueAsJsonNode.toString());
+
+                    bodyOrFormParams.add(k6Parameter);
+                }
+
+            } catch (JsonProcessingException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    /**
+     * Calculate order for this current request
+     *
+     * @param operationGroupingOrder
+     * @param requestsSize
+     * @return request order
+     */
+    private Integer calculateRequestOrder(OptionalInt operationGroupingOrder, int requestsSize) {
+        int requestOrder;
+
+        if (operationGroupingOrder.isPresent()) {
+            requestOrder = operationGroupingOrder.getAsInt() - 1;
+
+        } else {
+            switch (requestsSize) {
+            case 0:
+            case 1:
+                requestOrder = requestsSize;
+                break;
+
+            default:
+                requestOrder = (requestsSize - 1);
+                break;
+            }
+        }
+
+        return requestOrder;
+    }
+
+    //
+    /**
+     * Any variables not defined yet but used for subsequent data extraction must be
+     * initialized
+     *
+     * @param dataExtractSubstituteParams
+     * @param requestGroup
+     */
+    private void initializeDataExtractSubstituteParameters(
+            Map<String, DataExtractSubstituteParameter> dataExtractSubstituteParams, HTTPRequestGroup requestGroup) {
+
+        if (!dataExtractSubstituteParams.isEmpty()) {
+            List<String> existingVariablesNames = requestGroup.variables.stream().map(v -> v.key)
```

Comment:
```
List<String> existingVariablesNames = requestGroup.variables.stream().map(v -> v.key)
      .collect(Collectors.toList());

Set<DataExtractSubstituteParameter> initializeVariables = dataExtractSubstituteParams.values().stream()
     .filter(p -> !existingVariablesNames.contains(toVarName(p.paramName))).collect(Collectors.toSet());
```

this filter condition will O(1) fast if `existingVariablesNames` is a `Set`, when existingVariablesNames is a `List` filter will be O(n)

Reply:
Addressed.